### PR TITLE
Adding cloud run image processing tutorial

### PR DESCRIPTION
**Pull request by dinagraves**
Creating python sample for the [Cloud Run Image Processing Tutorial](https://cloud.google.com/run/docs/tutorials/image-processing#run_imageproc_deps-nodejs).

Most files should be diffed against the [Cloud Run PubSub sample](https://github.com/GoogleCloudPlatform/python-docs-samples/tree/master/run/pubsub).

`image.py` and `image_test.py` should be diffed against `main.py` and `main_test.py` from the [Image Magick](https://github.com/GoogleCloudPlatform/python-docs-samples/tree/master/functions/imagemagick) sample.

### CODE REVIEW

File: run/image-processing/image.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    with Image(filename=temp_local_filename) as image:
+        image.resize(*image.size, blur=16, filter='hamming')
+        image.save(filename=temp_local_filename)
+
+    print(f'Image {file_name} was blurred.')
+
+    # Upload result to a second bucket, to avoid re-triggering the function.
+    # You could instead re-upload it to the same bucket + tell your function
+    # to ignore files marked as blurred (e.g. those with a "blurred" prefix)
+    blur_bucket_name = os.getenv('BLURRED_BUCKET_NAME')
```
Comment (averikitsch) [inline run/image-processing/image.py]: I would expect to see this near the top of the file

File: run/image-processing/image.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    file_name = file_data['name']
+    bucket_name = file_data['bucket']
+
+    blob = storage_client.bucket(bucket_name).get_blob(file_name)
+    blob_uri = f'gs://{bucket_name}/{file_name}'
+    blob_source = {'source': {'image_uri': blob_uri}}
+
+    # Ignore already-blurred files
+    if file_name.startswith('blurred-'):
```
[the remark this answers is not in the document]
Comment (dinagraves) [reply run/image-processing/image.py]: I think this function is supposed to stay as close as possible to this one here: https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/functions/imagemagick/main.py

I think it's OK to keep it - someone could use the same bucket for input and output.  As long as they specify the environment variable, it'll still work.  

@grayside WDYT?  Should we update the function for our specific use case or keep it identical to the imagemagick one?

File: run/image-processing/image.py (hunk 1 above)
Comment (dinagraves) [reply run/image-processing/image.py]: Ditto above.

File: run/image-processing/.gcloudignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+.pytest_cache
```
[the remark this answers is not in the document]
Comment (dinagraves) [reply run/image-processing/.gcloudignore]: Do you mean I should remove the whole .gcloudignore file? Or do you mean that .pytest_cache can be removed from .gcloudignore?

File: run/image-processing/.gitignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+__pycache__
```
[the remark this answers is not in the document]
Comment (dinagraves) [reply run/image-processing/.gitignore]: Same question - should I remove .gitignore? Or just pycache from .gitignore?

File: run/image-processing/.gcloudignore (hunk 3 above)
[the remark this answers is not in the document]
Comment (averikitsch) [reply run/image-processing/.gcloudignore]: Sorry, we don't usually include the .gcloudignore file

File: run/image-processing/.gitignore (hunk 4 above)
[the remark this answers is not in the document]
Comment (averikitsch) [reply run/image-processing/.gitignore]: Same here, we shouldn't need the .gitignore file

File: run/image-processing/image.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (averikitsch) [reply run/image-processing/image.py]: You're right, we should probably keep it identical.

File: run/image-processing/README.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
+# Cloud Run Image Processing Sample
+
+This sample service applies [Cloud Storage](https://cloud.google.com/storage/docs)-triggered image processing with [Cloud Vision API](https://cloud.google.com/vision/docs) analysis and ImageMagick transformation.
+
+Use it with the [Image Processing with Cloud Run tutorial](http://cloud.google.com/run/docs/tutorials/image-processing).
+
+[![Run in Google Cloud][run_img]][run_link]
+
+[run_img]: https://storage.googleapis.com/cloudrun/button.svg
+[run_link]: https://console.cloud.google.com/cloudshell/editor?shellonly=true&cloudshell_image=gcr.io/cloudrun/button&cloudshell_git_repo=https://github.com/GoogleCloudPlatform/python-docs-samples&cloudshell_working_dir=run/image-processing
```
[the remark this answers is not in the document]
Comment (dinagraves) [reply run/image-processing/README.md]: Updated!

File: run/image-processing/README.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+```
+# Set an environment variable with your GCP Project ID
+export GOOGLE_CLOUD_PROJECT=<PROJECT_ID>
+
+# Submit a build using Google Cloud Build
+gcloud builds submit --tag gcr.io/${GOOGLE_CLOUD_PROJECT}/pubsub-tutorial
+
+# Deploy to Cloud Run
+gcloud beta run deploy pubsub-tutorial --image gcr.io/${GOOGLE_CLOUD_PROJECT}/pubsub-tutorial
```
[the remark this answers is not in the document]
Comment (dinagraves) [reply run/image-processing/README.md]: Added!

File: run/image-processing/README.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+```
+# Set an environment variable with your GCP Project ID
+export GOOGLE_CLOUD_PROJECT=<PROJECT_ID>
+
+# Submit a build using Google Cloud Build
+gcloud builds submit --tag gcr.io/${GOOGLE_CLOUD_PROJECT}/pubsub-tutorial
+
+# Deploy to Cloud Run
+gcloud beta run deploy pubsub-tutorial --image gcr.io/${GOOGLE_CLOUD_PROJECT}/pubsub-tutorial --set-env-vars=BLURRED_BUCKET_NAME=BLURRED_BUCKET_NAME
```
Comment (averikitsch) [inline run/image-processing/README.md]:
```suggestion
gcloud beta run deploy pubsub-tutorial --image gcr.io/${GOOGLE_CLOUD_PROJECT}/pubsub-tutorial --set-env-vars=BLURRED_BUCKET_NAME=<BLURRED_BUCKET_NAME>
```

**Pull request merged**